Review: Moveit doc (#10)
sangteak601/mujoco_ros2_control: doc/moveit_doc.rst

How To Setup mujoco_ros2_control
================================

Introduction
------------

This documents provide guidence on how to setup and use mujoco_ros2_control package.

> This page will demonstrate package's features with a robot manipulator, but the package is not specifically designed for manipulation and can be used for any types of robots.

Reviewer: I will apply grammar checker next time.

Prerequisites
--------------

Running on the local machine
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

If you want to run an application on the local machine, you need following dependencies installed on the machine.

> 1. MuJoCo

Reviewer: I will add them

> 2. ROS2 humble?

Reviewer: I haven't tested on the other distro.

> 3. MoveIt2

Reviewer: done

Running in the container
^^^^^^^^^^^^^^^^^^^^^^^^
If you want to run an application in docker container, you need following dependencies installed on the machine.

1. docker

Installation
------------

Running on the local machine
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1. Clone the repository.
  .. code-block:: bash

    git clone https://github.com/sangteak601/mujoco_ros2_control.git

2. Set environment variable
  .. code-block:: bash

    export MUJOCO_DIR=/PATH/TO/MUJOCO/mujoco-3.x.x

3. Build the package
  .. code-block:: bash

    cd mujoco_ros2_control
    source /opt/ros/${ROS_DISTRO}/setup.bash
    colcon build

Running in the container
^^^^^^^^^^^^^^^^^^^^^^^^

TODO

How to Setup mujoco_ros2_control
--------------------------------

.. note:: Please refer to `this <https://github.com/sangteak601/mujoco_ros2_control/blob/moveit_doc/doc/index.rst#usage>`_ for the details.

Set ros2_control plugin in the URDF
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> You can specify ros2_plugin in the URDF file as follows:

Author: ```suggestion
You can configure the `ros2_control` hardware system plugin for MuJoCO in the URDF file as follows:
```

.. code-block:: XML

  <ros2_control name="MujocoSystem" type="system">
    <hardware>
      <plugin>mujoco_ros2_control/MujocoSystem</plugin>
    </hardware>
  </ros2_control>

You can also set parameters for the plugin such as minimum joint position, maximum joint position and so on.
To find examples of parameters, please see `urdf examples <https://github.com/sangteak601/mujoco_ros2_control/tree/moveit_doc/mujoco_ros2_control_demos/urdf>`_.

Create MJCF(MuJoCo xml format)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

You need to convert the URDF model to a MJCF XML file.
Make sure to use the **same name** for the link and joint, which are mapped to the body and joint in Mujoco.

> You don't need to specify <limit> in MJCF because it will be taken care of in the plugin.

Author: Will it override if it is specified?

Reviewer: I think the smaller one will be applied.

> For force torque sensor, you need to map the sensor to a force sensor and a torque sensor in MJCF, since there is no combined force torque sensor in MuJoCo.

Author: ```suggestion
Any force torque sensors need to be mapped to separate force and torque sensors in the MJCF, since there is no support for combined sensors in MuJoCo.
```

Reviewer: done
